GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/NotifyUsersAndRetireInactiveAccountsJob.cs

```diff
@@ -0,0 +1,120 @@
+        {
+            this.dataRepository = dataRepository;
+            this.emailSendingService = emailSendingService;
+        }
+
+        public void NotifyUsersAndRetireInactiveAccounts()
+        {
+            var runId = JobHelpers.CreateRunId();
+            var startTime = VirtualDateTime.Now;
+            JobHelpers.LogFunctionStart(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+
+            try
+            {
+                SendReminderEmails();
+
+                RetireUsers();
+
+                JobHelpers.LogFunctionEnd(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime);
+            }
+            catch (Exception ex)
+            {
+                JobHelpers.LogFunctionError(runId, nameof(NotifyUsersAndRetireInactiveAccounts), startTime, ex);
+
+                //Rethrow the error
+                throw;
+            }
+        }
+
+        private void SendReminderEmails()
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+            DateTime threeYearsAgoMinusThirtyDays = threeYearsAgo.AddDays(30);
+            DateTime threeYearsAgoMinusSevenDays = threeYearsAgo.AddDays(7);
+
+            List<User> usersToSendReminders = dataRepository.GetAll<User>()
+                .Where(u => u.LoginDate == threeYearsAgoMinusThirtyDays || u.LoginDate == threeYearsAgoMinusSevenDays)
+                .ToList();
+
+            foreach (User user in usersToSendReminders)
+            {
+                // TODO: Update loginUrl
+                string daysRemaining = user.LoginDate == threeYearsAgoMinusThirtyDays ? "30" : "7";
+                emailSendingService.SendAccountRetirementNotificationEmail(user.EmailAddress, daysRemaining, "example url");
+            }
+
+        }
+
+        private void RetireUsers()
+        {
+            DateTime threeYearsAgo = VirtualDateTime.Now.AddYears(-3);
+
+            List<User> usersToRetire = dataRepository.GetAll<User>()
+                .Where(u => u.LoginDate < threeYearsAgo)
+                .Where(u => !u.HasBeenAnonymised)
+                .ToList();
+
+            foreach (User user in usersToRetire)
+            {
+                user.Firstname = $"User{user.UserId}";
+                user.Lastname = $"User{user.UserId}";
+                user.JobTitle = "Anonymised";
+                user.EmailAddress = "Anonymised";
+                user.ContactFirstName = "Anonymised";
+                user.ContactLastName = "Anonymised";
+                user.ContactJobTitle = "Anonymised";
+                user.ContactOrganisation = "Anonymised";
+                user.ContactEmailAddress = "Anonymised";
+                user.ContactPhoneNumber = "Anonymised";
+                user.Salt = "Anonymised";
+                user.PasswordHash = "Anonymised";
+
+                user.HasBeenAnonymised = true;
+                user.Status = UserStatuses.Retired;

```

Comment:
Instead of:
```csharp
user.Status = UserStatuses.Retired;
```
there is a method that does this (as well as adding an entry to the UserStatus table (which I wouldn't have expected you to be aware of!)

Something like:
```csharp
user.SetStatus(UserStatuses.Retired, user, "User retired by RetireInactiveAccountsJob");
```